lightning/src/routing/scoring.rs

```diff
@@ -756,6 +793,50 @@ impl<G: Deref<Target = NetworkGraph<L>>, L: Deref, T: Time> ProbabilisticScorerU
 		None
 	}
 
+	/// Query the historical estimated minimum and maximum liquidity available for sending a
+	/// payment over the channel with `scid` towards the given `target` node.
+	///
+	/// Returns two sets of 8 buckets. The first set describes the octiles in which we've seen our
+	/// lower-bound liquidity estimates, the second set describes the octiles in which we've seen
+	/// our upper-bound liquidity estimates. Each bucket describes the relative frequency at which
+	/// we've seen a liquidity bound in the octile relative to the channel's total capacity, on an
+	/// arbitrary scale. Because the values are slowly decayed, more recent data points count for
+	/// more than older datapoints.
+	///
+	/// When scoring, the estimated probability that an upper-/lower-bound lies in a given octile
+	/// relative to the channel's total capacity is calculated by dividing that bucket's value with
+	/// the total of all buckets for the given bound.
```

Comment:
Would it be useful to provide guarantee/safety that total of all buckets will not be equal to zero?

Reply:
It is possible that all the buckets could be equal to zero at some point however right? I.e. in the case that all the buckets had decayed to 0? Either way you're right that that's probably something that would be helpful to indicate to the user

Reply:
Yes i understand, in that case we might want to add a "Caution:". (even though clients shouldn't be assuming this)